assume/markets/clearing_algorithms/complex_clearing.py

```diff
@@ -136,114 +136,128 @@ def energy_balance_rule(m, t):
     return instance, results
 
 
-def pay_as_clear_complex(
-    market_agent: MarketRole,
-    market_products: list[MarketProduct],
-):
-    """
-    This implements pay-as-clear with more complex bid structures, including acceptance ratios, bid types, and profiled volumes.
+class ComplexClearingRole(MarketRole, MarketMechanism):
+    def __init__(self, marketconfig: MarketConfig):
+        super().__init__(marketconfig)
+        assert "bid_type" in self.marketconfig.additional_fields
 
-    :param market_agent: The market agent
-    :type market_agent: MarketRole
-    :param market_products: The products to be traded
-    :type market_products: list[MarketProduct]
+    def validate_orderbook(self, orderbook: Orderbook, agent_tuple) -> None:
```

Comment:
`bid_type` does not exist in the core market mechanism

Reply:
I mean one can add if "bid_type" in self.marketconfig.additional_field. otherwise we also need to copy all the price and other checks into the complex clearing all other future ones we will be adding.

Reply:
we are calling: `super().validate_orderbook(orderbook, agent_tuple)` which validates everything from the base class as well?
Is this what you mean?

Reply:
oh now I see, sorry for the confusion